packages/interactivity/src/vdom.ts

```diff
@@ -38,33 +40,49 @@ export const hydratedIslands = new WeakSet();
 /**
  * Recursive function that transforms a DOM tree into vDOM.
  *
- * @param {Node} root The root element or node to start traversing on.
- * @return {import('preact').VNode[]} The resulting vDOM tree.
+ * @param root The root element or node to start traversing on.
+ * @return The resulting vDOM tree.
  */
-export function toVdom( root ) {
+export function toVdom( root: Node ): [ string | VNode | null, Node | null ] {
```

Comment:
I'm not happy with the changes here…

Reply:
`toVdom` is an internal function. So TS here is not critical. Don't let this block your PR 😄